OvmfPkg/Library/PlatformInitLib/Platform.c

```diff
@@ -801,8 +813,24 @@ PlatformValidateNvVarStore (
       (NvVarStoreFvHeader->FvLength != NvVarStoreSize)
       )
   {
-    DEBUG ((DEBUG_ERROR, "NvVarStore FV headers were invalid.\n"));
-    return FALSE;
+    if (!Retry) {
+      DEBUG ((DEBUG_ERROR, "NvVarStore FV headers were invalid.\n"));
+      return FALSE;
+    }
+
+    DEBUG ((DEBUG_INFO, "Remapping NvVarStore as shared\n"));
+    Status = MemEncryptSevClearMmioPageEncMask (
+               0,
+               (UINTN)NvVarStoreBase,
+               EFI_SIZE_TO_PAGES (NvVarStoreSize)
+               );
+    if (EFI_ERROR (Status)) {

```

Comment:
QemuFlashFvbServicesRuntimeDxe does the same thing in `MarkIoMemoryRangeForRuntimeAccess()`. 
I guess that can be dropped?

Reply:
I think those are two different page tables, right, PEI and DXE? The DXE phase will create new page tables, so I think that still needs to be there since PlatformInitLib is only used as part of PEI.

Reply:
Yes, we switch away from the initial page tables at some point.  I think you are right.